src/interpreter/util/modifier/has-modifier.ts

```diff
@@ -3,7 +3,20 @@ import {TS} from "../../../type/ts.js";
 /**
  * Returns true if the given Node has the given kind of Modifier
  */
-export function hasModifier(node: TS.Node | TS.Modifier[], modifier: TS.Modifier["kind"]): boolean {
-	const modifiers = Array.isArray(node) ? node : (node.modifiers as readonly TS.Modifier[] | undefined);
-	return modifiers != null && modifiers.some(m => m.kind === modifier);
+export function hasModifier(node: TS.Node | TS.Modifier[], modifier: TS.Modifier["kind"], typescript: typeof TS): boolean {
```

Comment:
I would prefer a much simpler implementation where we just check for the existence of an array of modifiers on the Node to avoid having to pass in a reference to the TypeScript object and using the new `canHaveModifiers` helper.

The existing implementation could simply be rewritten to this, which will work in TS5 as well:

```ts
/**
 * Returns true if the given Node has the given kind of Modifier
 */
export function hasModifier(node: TS.Node | TS.ModifierLike[], modifier: TS.Modifier["kind"]): boolean {
	const modifiers = Array.isArray(node) ? node : "modifiers" in node && Array.isArray(node.modifiers) ? (node.modifiers as readonly TS.ModifierLike[]) : undefined;
	return modifiers != null && modifiers.some(m => m.kind === modifier);
}
```